### ISSUE TITLE
Uncaught TypeError: Cannot create property 'pending' on boolean 'true'

### ISSUE THREAD

**Issue by sylph01** [atom/tree-view#1080]
### Steps to reproduce

1. With no active editor tabs open
    * = No items are open in any tab, including "Untitled"
2. Right click on tree view item, "Split Right"
    * "Split Up" "Split Down" "Split Left" did the same too

**Atom**: 1.16.0 x64
**Electron**: 1.3.13
**OS**: Mac OS X 10.12.4
**Thrown From**: [tree-view](https://github.com/atom/tree-view) package 0.215.1

### Stack Trace

Uncaught TypeError: Cannot create property 'pending' on boolean 'true'

```
At /Applications/Atom.app/Contents/Resources/app.asar/src/workspace.js:525

TypeError: Cannot create property 'pending' on boolean 'true'
    at Workspace.open (/app.asar/src/workspace.js:525:23)
    at TreeView.module.exports.TreeView.openAfterPromise (/app.asar/node_modules/tree-view/lib/tree-view.js:497:31)
    at TreeView.module.exports.TreeView.openSelectedEntry (/app.asar/node_modules/tree-view/lib/tree-view.js:826:21)
    at TreeView.module.exports.TreeView.openSelectedEntrySplit (/app.asar/node_modules/tree-view/lib/tree-view.js:842:23)
    at TreeView.module.exports.TreeView.openSelectedEntryRight (/app.asar/node_modules/tree-view/lib/tree-view.js:848:19)
    at /app.asar/node_modules/tree-view/lib/tree-view.js:240:26)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/app.asar/src/command-registry.js:259:29)
    at CommandRegistry.handleCommandEvent (/app.asar/src/command-registry.js:3:59)
    at CommandRegistry.module.exports.CommandRegistry.dispatch (/app.asar/src/command-registry.js:160:19)
    at AtomEnvironment.module.exports.AtomEnvironment.dispatchContextMenuCommand (/app.asar/src/atom-environment.js:1185:28)
    at EventEmitter.outerCallback (/app.asar/src/application-delegate.js:325:25)
    at emitThree (events.js:116:13)
    at EventEmitter.emit (events.js:194:7)
```

### Commands

```
  3x -0:18.8.0 tree-view:open-selected-entry-right (span.name.icon.icon-file-text)
```

### Non-Core Packages

```
language-elixir 0.17.0 
markdown-themeable-pdf 1.2.0 
platformio-ide-terminal 2.2.3 
```

**Comment by 50Wliu** [comment]
I can't reproduce this on Atom 1.18.0-dev-1c2fbbe63, Windows 10 x64. /cc @ungb

**Comment by rsese** [comment]
I just tried on macOS 10.12.4 with Atom 1.16.0 but couldn't reproduce - @sylph01, can you reproduce this in safe mode (`atom --safe`)?

**Comment by sylph01** [comment]
I was able to reproduce this on Atom 1.16.0, also in safe mode.
(I tried the same on a different machine with 1.15.0, and in that case, it didn't reproduce)

I will ask some other people with similar environments to reproduce this issue.

----

[Enter steps to reproduce:]

1. Start Atom with `atom . --safe` (or `atom .`)
2. (Same as in the header)

**Atom**: 1.16.0 x64
**Electron**: 1.3.13
**OS**: Mac OS X 10.12.3
**Thrown From**: [tree-view](https://github.com/atom/tree-view) package 0.215.1


### Stack Trace

Uncaught TypeError: Cannot create property 'pending' on boolean 'true'

```
At /Applications/Atom.app/Contents/Resources/app.asar/src/workspace.js:525

TypeError: Cannot create property 'pending' on boolean 'true'
    at Workspace.open (/app.asar/src/workspace.js:525:23)
    at TreeView.module.exports.TreeView.openAfterPromise (/app.asar/node_modules/tree-view/lib/tree-view.js:497:31)
    at TreeView.module.exports.TreeView.openSelectedEntry (/app.asar/node_modules/tree-view/lib/tree-view.js:826:21)
    at TreeView.module.exports.TreeView.openSelectedEntrySplit (/app.asar/node_modules/tree-view/lib/tree-view.js:842:23)
    at TreeView.module.exports.TreeView.openSelectedEntryRight (/app.asar/node_modules/tree-view/lib/tree-view.js:848:19)
    at /app.asar/node_modules/tree-view/lib/tree-view.js:240:26)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/app.asar/src/command-registry.js:259:29)
    at CommandRegistry.handleCommandEvent (/app.asar/src/command-registry.js:3:59)
    at CommandRegistry.module.exports.CommandRegistry.dispatch (/app.asar/src/command-registry.js:160:19)
    at AtomEnvironment.module.exports.AtomEnvironment.dispatchContextMenuCommand (/app.asar/src/atom-environment.js:1185:28)
    at EventEmitter.outerCallback (/app.asar/src/application-delegate.js:325:25)
    at emitThree (events.js:116:13)
    at EventEmitter.emit (events.js:194:7)
```

### Commands

```
     -0:02.8.0 tree-view:open-selected-entry-right (span.name.icon.icon-file-text)
```

### Non-Core Packages

```

```

**Comment by sylph01** [comment]
[I've recorded the process of reproducing the error.](https://s01.ninja/files/tab.mp4) (Note: file placed under my web server) Hope this will help...

**Comment by ungb** [comment]
I was unable to repro this issue as well.  Probably not related but do you have any community packages installed?  you can find this by running `apm list` on terminal.

**Comment by sylph01** [comment]
Seems like I found the trigger:

* With "Allow Pending Pane Items" turned **off**, I get this error.
* With "Allow Pending Pane Items" turned **on**, this error does not happen.

All the machines I tested (including the one I recorded with) had this "Allow Pending Pane Items" turned off.

**Comment by sylph01** [comment]
I wrote a patch that fixes this issue. I'm new to the JS/Atom world, so it would help a lot if anyone could help me with the patch. Thank you!

**Comment by ungb** [comment]
Thanks @sylph01! I was able to repro the issue with your last comment!    I'll take a look at the PR tmrw and see if we can get it merged.

**Comment by ungb** [comment]
I was able to repro this before but looks fixed as 1.18 stable.  going to close this issue, please feel free to let us know  if you are still seeing this issue
